### PR TITLE
Make request handling logic pluggable.

### PR DESCRIPTION
**Pull request by efeg**
Addresses the issue https://github.com/linkedin/cruise-control/issues/770.

### CODE REVIEW

File: cruise-control/src/main/java/com/linkedin/kafka/cruisecontrol/servlet/parameters/AdminParameters.java [hunk 1]
```diff
@@ -32,13 +32,12 @@
   private Integer _concurrentInterBrokerPartitionMovements;
   private Integer _concurrentLeaderMovements;
   private Integer _reviewId;
-  private final boolean _twoStepVerificationEnabled;
+  private boolean _twoStepVerificationEnabled;
```
Comment (kidkun) [inline cruise-control/src/main/java/com/linkedin/kafka/cruisecontrol/servlet/parameters/AdminParameters.java]: (Not introduce by this patch)this field is one-time use(in `initParameters()`), looks like we do not need to store it in a field, just like what `RebalanceParameters` does. also apply to `PauseResumeParameters`/`StopProposalParameters`.

File: cruise-control/src/main/java/com/linkedin/kafka/cruisecontrol/servlet/handler/async/AddBrokerRequest.java [hunk 2]
```diff
@@ -5,16 +5,17 @@
...
 
 
 public class AddBrokerRequest extends AbstractAsyncRequest {
-  private final AddBrokerParameters _parameters;
+  private AddBrokerParameters _parameters;
 
-  public AddBrokerRequest(KafkaCruiseControlServlet servlet, AddBrokerParameters parameters) {
-    super(servlet);
-    _parameters = parameters;
+  public AddBrokerRequest() {
```
Comment (kidkun) [inline cruise-control/src/main/java/com/linkedin/kafka/cruisecontrol/servlet/handler/async/AddBrokerRequest.java]: maybe we do not need to explicitly specify default constructor here(apply to other `Request` class)?

File: cruise-control-core/src/main/java/com/linkedin/cruisecontrol/servlet/EndPoint.java [hunk 3]
```diff
@@ -0,0 +1,16 @@
...
+
+package com.linkedin.cruisecontrol.servlet;
+
+/**
+ * An enum that lists all supported endpoints by Cruise Control.
+ */
+public interface EndPoint {
+
+  /**
+   * @return The endpoint type, which might be relevant for the application logic -- e.g. selective response caching.
```
[the remark this answers is not in the document]
Comment (efeg) [reply cruise-control-core/src/main/java/com/linkedin/cruisecontrol/servlet/EndPoint.java]: Made it more verbose.

File: cruise-control/src/main/java/com/linkedin/kafka/cruisecontrol/servlet/parameters/AddBrokerParameters.java [hunk 4]
```diff
@@ -28,8 +28,8 @@
 public class AddBrokerParameters extends AddedOrRemovedBrokerParameters {
   private boolean _throttleAddedBrokers;
 
-  public AddBrokerParameters(HttpServletRequest request, KafkaCruiseControlConfig config) {
-    super(request, config);
+  public AddBrokerParameters() {
```
[the remark this answers is not in the document]
Comment (efeg) [reply cruise-control/src/main/java/com/linkedin/kafka/cruisecontrol/servlet/parameters/AddBrokerParameters.java]: To create an instance of such subclasses using `KafkaCruiseControlConfig#getConfiguredInstance(String, Class<T>, Map<String, Object>`, or any other methods that call `Utils#newInstance(Class<T>)`, the class needs to explicitly specify a public no-argument constructor.

File: cruise-control/src/main/java/com/linkedin/kafka/cruisecontrol/servlet/handler/async/AddBrokerRequest.java (hunk 2 above)
Comment (efeg) [reply cruise-control/src/main/java/com/linkedin/kafka/cruisecontrol/servlet/handler/async/AddBrokerRequest.java]: To create an instance of such subclasses using `KafkaCruiseControlConfig#getConfiguredInstance(String, Class<T>, Map<String, Object>`, or any other methods that call `Utils#newInstance(Class<T>)`, the class needs to explicitly specify a public no-argument constructor.

**Pull request merged**